dias/analyzers/sensitivity_analyzer.py

```diff
@@ -0,0 +1,374 @@
+    cyl_start_num = config.Property(proptype=int, default=2)
+    lag = config.Property(proptype=str2timedelta, default="4h")
+
+    def run(self):
+        """Task stage: analyzes data from the last period."""
+        stop_time = datetime.utcnow() - self.lag
+        # Query files from now to period hours back
+        start_time = stop_time - self.period
+
+        # Find all calibration files
+        f = self.Finder()
+        f.set_time_range(start_time, stop_time)
+        f.accept_all_global_flags()
+        f.only_corr()
+        f.filter_acqs((data_index.ArchiveInst.name == self.acq_suffix))
+        file_list = f.get_results()
+        all_files = file_list[0][0]
+
+        if not all_files:
+            err_msg = 'No {} files found from last {}.'.format(
+                self.acq_suffix, self.period)
+            self.logger.error(err_msg)
+            raise exception.DiasDataError(msg)
+
+        self.logger.info('Calculating sensitivity from %s to %s' %
+                         (str(start_time), str(stop_time)))
+
+        # Get Unix time for the start time for timestamp
+        time_tuple = start_time.timetuple()
+        start_time_unix = calendar.timegm(time_tuple)
+        timestamp0 = start_time_unix
+
+        # Look up inputmap
+        inputmap = tools.get_correlator_inputs(
+            ephemeris.unix_to_datetime(timestamp0),
+            correlator=self.correlator)
+
+        # Read a sample file for getting index map
+        file_sample = all_files[0]
+        data = andata.CorrData.from_acq_h5(file_sample,
+                                           datasets=[
+                                               'reverse_map', 'flags/inputs'],
+                                           apply_gain=False, renormalize=False)
+
+        # Get baselines
+        prod, prodmap, dist, conj, cyl, scale = self.get_baselines(
+            data.index_map, inputmap)
+
+        for files in all_files:
+
+            # Load index map and reverse map
+            data = andata.CorrData.from_acq_h5(
+                files, datasets=['reverse_map', 'flags/inputs'],
+                apply_gain=False, renormalize=False)
+
+            # Determine axes
+            nfreq = data.nfreq
+            nblock = int(np.ceil(nfreq / float(self.nfreq_per_block)))
+
+            timestamp = data.time
+            ntime = timestamp.size
+
+            # Determine groups
+            polstr = np.array(sorted(prod.keys()))
+            npol = polstr.size
+
+            # Calculate counts
+            cnt = np.zeros(
+                (data.index_map['stack'].size, ntime), dtype=np.float32)
+
+            if np.any(data.flags['inputs'][:]):
+                for pp, ss in zip(data.index_map['prod'][:],
+                                  data.reverse_map['stack']['stack'][:]):
+                    cnt[ss, :] += (data.flags['inputs'][pp[0], :]
+                                   * data.flags['inputs'][pp[1], :])
```

Comment:
I profiled this analyzer and it turns out to spend about 50% of the time calling `flags` here (it doesn't look like, but it's a function). So we should see a massive perfomance improve if you save `data.flags['inputs']` outside of the loop.